libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -377,8 +377,9 @@ bool ExpressionCompiler::visit(BinaryOperation const& _binaryOperation)
 		if (Token::isCompareOp(c_op))
 			cleanupNeeded = true;
 		if (
-			(commonType.category() == Type::Category::Integer || commonType.category() == Type::Category::FixedPoint) &&
-			(c_op == Token::Div || c_op == Token::Mod)
+			(commonType.category() == Type::Category::Integer &&
+			(c_op == Token::Div || c_op == Token::Mod)) ||
+			commonType.category() == Type::Category::FixedPoint
```

Comment:
Are you sure about that? If yes, I think it would improve readability to move the fixed point case to its own if.